### DISCUSSION TITLE
How do you create and open child windows?

### DISCUSSION THREAD

**Discussion by javageek68** [alejandroautalan/pygubu-designer#72] [Q&A] [answered]
I've created two windows using Pygubu Designer.  The first is my main application.  The second is a settings window.  It allows the user to store settings to a file.  The intent is to allow the user to click a button in the main application window to open the settings window.  The settings window opens the settings file and populates each entry box via its text variable.  The user edits the settings and clicks a Close button.  Then, the settings window stores the settings to the file and closes itself.  So far, each of my attempts either opens the settings window in the same window as the parent or the text variable does not populate its Entry box when loading.  Here is my code.

    class DashGuiApp(DashGuiAppBase):
        def __init__(self, master=None):
            super(DashGuiApp, self).__init__(master)
            # init code ...    

        def edit_settings(self):
            """Create and open the settings window"""
            #self.open_settings_window_1()
            self.open_settings_window_2()
            #self.open_settings_window_3()
    
        def open_settings_window_1(self):
            """This creates a child window for the settings.  But the text variable doesn't populate its entry control"""
            settings_root = tk.Tk()
            # DashSettingsBackend contains the class I created using Pygubu Designer.  It contains a list
            # of Entry controls.  Each entry control has a text variable I use to load previously saved
            # settings from a file.
            settings_window = DashSettingsBackend(settings_root)
    
        def open_settings_window_2(self):
            """This allows the text variable to populate its entry control.  But it does not open settings
            in a seperate window.  Instead, it replaces the contents of the parent window."""
            settings_window = DashSettingsBackend(root)
    
        def open_settings_window_3(self):
            """This creates a seperate child window for the settings.  But I don't know how to set its 
            contents to DashSettingsBackend """
            top = tk.Toplevel(root)
            # If I uncomment this, it does not create a child window for the settings.  Instead, it 
            # replaces the contents of the parent window with the contents of the child windod.
            #top.geometry = DashSettingsBackend(app)
		    
    if __name__ == '__main__':
        """Create and open the main window"""
        root = tk.Tk()
        app = DashGuiApp(root)
        app.run()
	
Note - DashGuiAppBase is the parent class I created with Pygubu Desigher.  DashSettingsBackend is the child window I created with Pygubu Designer

What is the correct way to open a child window?

**Comment by alejandroautalan** [comment]
Hello @javageek68 thanks for trying pygubu.

There are several ways to create child windows. It will depend on how you want to display the child window, modal or not.
I need more info about your issue.
- How are you generating the classes? Using the template: 'Pygubu Application' or 'Code script'? Can you paste one example of your DashSettingsBackend?
- Can you show me the ui file or files?

In the mean time, I will prepare an example on how to create a simple app with a settings dialog.
Let me know

Regards
Alejandro A.

**Comment by alejandroautalan** [comment, marked as answer]
Added examples of [dialog](https://github.com/alejandroautalan/pygubu-designer/tree/master/examples/dialogs) windows.
